src/Form/SearchSolrForm.php

```diff
@@ -0,0 +1,240 @@
+<?php
+
+namespace Drupal\openy\Form;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Defines a form for setting Search Solr server parameters during install.
+ */
+class SearchSolrForm extends FormBase {
```

Comment:
Seems like ConfigFormBase is a way to go here
See https://github.com/ymcatwincities/openy/blob/8.x-2.x/modules/custom/openy_analytics/src/Form/SettingsForm.php#L11

Let's leverage Drupal core functionality and create less legacy code
with ConfigFormBase you'd have less code in submit and proper architecture

Thank you

Reply:
Fixed